app/models/host/managed.rb

```diff
@@ -452,13 +450,22 @@ def self.import_host_and_facts(hostname, facts, certname = nil, proxy_id = nil)
     # if we were given a certname but found the Host by hostname we should update the certname
     host.certname = certname if certname.present?
 
+    # if proxy authentication is enabled and we have no puppet proxy set, use it.
+    host.puppet_proxy_id ||= proxy_id
+
     # if proxy authentication is enabled and we have no puppet proxy set and the upload came from puppet,
     # use it as puppet proxy.
-    if facts['_type'].blank? || facts['_type'] == 'puppet'
+    if import_type == 'puppet'
       host.puppet_proxy_id ||= proxy_id
     end
 
-    host.save(:validate => false) if host.new_record?
+    host
```

Comment:
So that you can use this method when you want to trigger import_facts later on it's result, also I don't think the save is needed at this point, especially with validation skipping. If we'd keep same contract, the method would return only true or nil, so you couldn't get to newly built host. Maybe a small example will clarify it better

```ruby
h = Host.import_host(...) # class method, similar to Host.new
h.import_facts(...) # now we can call other methods on new instance
```

I'd say that important thing is that contract of import_host_and_facts hasn't changed, not that this is a new method, and git only shows it as part of old import_host_and_facts since the body is similar